### PR TITLE
HPCC-15319 #options following ws ignored in BEGINC++

### PR DESCRIPTION
**Pull request by jamienoss**
Signed-off-by: James Noss james.noss@lexisnexis.com

### CODE REVIEW

**Comment by jamienoss** [comment]
@ghalliday please review.

The runtime test doesn't actually work c.f. https://track.hpccsystems.com/browse/HPCC-15319 &  https://track.hpccsystems.com/browse/HPCC-15324

**Comment by ghalliday** [comment]
@jamienoss 
I don't really like the change as it stands for a few reasons
- I would expect #options to be on their own line
- this code would match printf(" #option ');
- the test case apparently doesn't work.

**Comment by richardkchapman** [comment]
It would be ok to allow leading whitespace but as Gavin says, you can't just check that the prior char is whitespace.

File: testing/regress/ecl/cppbody7.ecl [hunk 1]
```diff
@@ -19,21 +19,20 @@
...
 BEGINC++
+//default as 'action'
 return rtlRandom();
 ENDC++;
 
 integer4 mkRandom2 :=
 BEGINC++
-#option pure
-return rtlRandom();
+return rtlRandom(); #option pure
```
Comment (richardkchapman) [inline testing/regress/ecl/cppbody7.ecl]: This should give an error - as in C++ and ECL, the # must be at the start of the line (only preceding whitespace allowed)

**Comment by richardkchapman** [comment]
I tested the existing code, and it is behaving as I would expect - accepting #option at the start of a line with optional leading whitespace, but nowhere else.

**Pull request closed without merging**